proxy/http/HttpTransactHeaders.cc

```diff
@@ -1073,3 +1073,60 @@ HttpTransactHeaders::remove_privacy_headers_from_request(HttpConfigParams *http_
     }
   }
 }
+
+void
+HttpTransactHeaders::normalize_accept_encoding(const OverridableHttpConfigParams *ohcp, HTTPHdr *header)
+{
+  int normalize_ae = ohcp->normalize_ae;
+
+  if (normalize_ae == 99) { // Not explicitly configured.
+
+    if (ohcp->normalize_ae_gzip == 99) { // Not explicitly configured.
+
+      normalize_ae = 1; // Enable normalization to gzip by default.
+
+    } else {
+      static bool logged = false;
+
+      if (!logged) {
+        Note("proxy.config.http.normalize_ae_gzip depricated, use proxy.config.http.normalize_ae");
+        logged = true;
+      }
+
+      normalize_ae = ohcp->normalize_ae_gzip ? 1 : 0;
+    }
+  }
+
+  if (normalize_ae) {
+    MIMEField *ae_field = header->field_find(MIME_FIELD_ACCEPT_ENCODING, MIME_LEN_ACCEPT_ENCODING);
+
+    if (ae_field) {
+      if (normalize_ae == 1) {
+        // Force Accept-Encoding header to gzip or no header.
+        if (HttpTransactCache::match_content_encoding(ae_field, "gzip")) {
+          header->field_value_set(ae_field, "gzip", 4);
+          Debug("http_trans", "[Headers::normalize_accept_encoding] normalized Accept-Encoding to gzip");
+        } else {
+          header->field_delete(ae_field);
+          Debug("http_trans", "[Headers::normalize_accept_encoding] removed non-gzip Accept-Encoding");
+        }
+      } else if (normalize_ae == 2) {
```

Comment:
So, this doesn't really work as we need to. With set to "2", it must still normalize on gzip if the client did not AE: br. Otherwise, we'll break all previous behavior of normalizing on gzip.